vizro-core/examples/_chart-gallery/utils/custom_extensions.py

```diff
@@ -0,0 +1,116 @@
+"""Contains custom components and charts used inside the dashboard."""
+
+from typing import List, Literal
+
+import dash_bootstrap_components as dbc
+import pandas as pd
+import plotly.graph_objects as go
+import vizro.models as vm
+from dash import dcc, html
+from vizro.models.types import capture
+
+try:
+    from pydantic.v1 import Field
+except ImportError:  # pragma: no cov
+    from pydantic import Field
+
+
+class CodeClipboard(vm.VizroBaseModel):
+    """Contains code snippet with a copy to clipboard button."""
+
+    type: Literal["code_clipboard"] = "code_clipboard"
+    title: str = "Code"
+    text: str
+
+    def build(self):
+        """Returns the code clipboard component inside an accordion."""
+        return dbc.Accordion(
+            [
+                dbc.AccordionItem(
+                    dbc.Card(
+                        [
+                            html.H3(self.title),
+                            dcc.Markdown(self.text, id=self.id, className="code-block"),
+                            dcc.Clipboard(target_id=self.id, className="code-clipboard"),
+                        ]
+                    ),
+                    title="SHOW CODE",
+                )
+            ],
+            start_collapsed=True,
```

Comment:
I also found it odd that this was collapsed by default or even collapsible at all, since it leaves a big empty space on the bottom right of the screen. I don't know whether the idea is that we might have multiple code snippets/examples in one page in future and it would work better then? But probably in that case I would just put it in tabs and nest one example per tab, with each tab having a graph on the left and code on the right like you have now.

Reply:
I think it really depends on the user group. My idea was to hide information that might not be relevant to everyone. If you look at other dashboard chart examples, they usually start with code snippets collapsed as well. This helps avoid overwhelming users with too much content, especially if engineers aren't the only users. For mobile, you don't want the entire code visible right away—they might not even open it there.

Another reason is scrolling. Some code snippets are long and can make the page scroll immediately, which I personally didn't like.

Initially, I used tabs—one for the chart and another for the code. However, I found it more useful to have the chart and code side-by-side, as @antonymilne suggested. Empty space isn’t generally bad 😄. Even without the code snippet, I’d leave some space empty because of the chart ratio. If the chart takes the entire width, it can look odd for some charts (e.g., pie charts aren't suited for rectangular shapes).

I also experimented with the layout, placing text on the left, and the chart and code snippet on the right, moving things around quite a bit 😄. But I couldn't find a layout that didn’t look odd when the code snippet was expanded or if the chart's aspect ratio wasn’t nice. This is the setup that I found the least bothersome. But I'm happy for you guys to experiment with this! I might have been looking at this for too long 😄

Reply:
J. prefers to have it uncollapsed as well 😄  So maybe we just ignore mobile and scrolling etc. - I won't optimize for this here. 

He mentioned something similar to you @antonymilne in terms of having multiple tabs of code snippets - one chart and one chart inside the dashboard. Not sure though what the default should be etc. Let's just quickly discuss in our catch-up 👍